src/components/ChatApp/SearchField.react.js

```diff
@@ -64,12 +67,26 @@ class ExpandingSearchField extends Component {
     this.props.onTextChange(event);
   };
 
-  onClickRecent = () => {
+  onClickRecent = (state, props) => {
+    const { searchCount } = this.props;
+
+    const { indexCnt } = this.state;
+
     this.props.scrollRecent();
+    this.setState({
+      indexCnt: (indexCnt % searchCount) + 1,
+    });
   };
 
-  onClickPrev = () => {
+  onClickPrev = (state, props) => {
+    const { indexCnt } = this.state;
+
     this.props.scrollPrev();
+    if (indexCnt > 1) {
+      this.setState({ indexCnt: indexCnt - 1 });
```

Comment:
To fix the up arrow issue make changes here, 
Hint: If indexCnt goes below a certain value, change it back to searchCount. Some modification might be need but it should be pretty straight forward.

Reply:
i got what you want but the problem is the counter value is changing but its not highlighting the words when i press the uparrow, it has some other issue

Reply:
i got a fix